app/controllers/shared/api.php

```diff
@@ -49,44 +48,242 @@
+               ;
+
+            if ($event === Database::EVENT_DOCUMENT_DELETE) {
+                //Project documents deduction
+                $dbDocuments      = $dbForProject->getDocument('stats', md5("_inf_" . "{$document['databaseId']}" . ".documents"));
+                $projectDocuments = $dbForProject->getDocument('stats', md5("_inf_documents"));
+                if (!$dbDocuments->isEmpty()) {
+                    $dbForProject->decreaseDocumentAttribute(
+                        'stats',
+                        $projectDocuments->getId(),
+                        'value',
+                        $dbDocuments['value']
+                    );
+                }
+            }
+            break;
+
+        case str_starts_with($document->getCollection(), 'database_') && str_contains($document->getCollection(), '_collection_'): //documents
+            $queueForUsage
+                ->addMetric("documents", $value)  // per project
+                ->addMetric("{$document->getAttribute('$databaseId')}" . ".documents", $value) // per database
+                ->addMetric("{$document->getAttribute('$databaseId')}" . "." . "{$collection->getId()}" . ".documents", $value)  // per collection
+                ;
+            break;
+        case $document->getCollection() === 'buckets':
+            $queueForUsage->addMetric("buckets", $value); // per project
+
+            if ($event === Database::EVENT_DOCUMENT_DELETE) {
+                //Project files deduction
+                $bucketFiles  = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".files"));
+                $projectFiles = $dbForProject->getDocument('stats', md5("_inf_files"));
+                if (!$bucketFiles->isEmpty()) {
+                    $dbForProject->decreaseDocumentAttribute(
+                        'stats',
+                        $projectFiles->getId(),
+                        'value',
+                        $bucketFiles['value']
+                    );
+                }
+                //Project files storage deduction
+                $bucketStorage  = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".files.storage"));
+                $projectStorage = $dbForProject->getDocument('stats', md5("_inf_files.storage"));
+                if (!$bucketStorage->isEmpty()) {
+                    $dbForProject->decreaseDocumentAttribute(
+                        'stats',
+                        $projectStorage->getId(),
+                        'value',
+                        $bucketStorage['value']
+                    );
                 }
             }
             break;
+        case str_starts_with($document->getCollection(), 'bucket_'): // files
+            $queueForUsage
+                ->addMetric("files", $value) // per project
+                ->addMetric("files.storage", $document->getAttribute('sizeOriginal') * $value) // per project
```

Comment:
Should we also have a `storage` metric that will account for `files.storage` + `deployments.storage` + ... later (build.storage, statichosting.stroage) and such? or do we calculate that during runtime in billing?

Reply:
@eldadfux suggested, let's keep it separate it's fine to do this on the billing itself